addons/mail/static/src/js/text_emojis_field.js

```diff
@@ -0,0 +1,95 @@
+odoo.define('social.form_field_emoji', function (require) {
+"use strict";
+
+var basicFields = require('web.basic_fields');
+var core = require('web.core');
+var emojis = require('mail.emojis');
+var registry = require('web.field_registry');
+var MailEmojisMixin = require('mail.emoji_mixin');
+var QWeb = core.qweb;
+
+/*
+ * Common code for FieldText and FieldChar
+ */
+var FieldEmojiCommon = {
+    /**
+     * @override
+     * @private
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.emojis = emojis;
+    },
+
+    /**
+     * This will add an emoji button that shows the emojis selection dropdown.
+     *
+     * We use 'on_attach_callback' because we need the element to be attached to the form first.
+     * That's because the $emojisIcon element needs to be rendered outside of this $el
+     * (which is an textarea, that can't 'contain' any other elements).
+     *
+     * @override
+     */
+    on_attach_callback: function () {
+        if (!this.$emojisIcon) {
+            this.$emojisIcon = $(QWeb.render('mail.EmojisDropdown', {widget: this}));
+            this.$emojisIcon.find('.o_mail_emoji').on('click', (ev) => {
+                this._onEmojiClick(ev);
+                this._isDirty = true;
+            });
+            this.$el.after(this.$emojisIcon);
+        }
+
+        if (this.mode === 'edit') {
+            this.$emojisIcon.show();
+        } else {
+            this.$emojisIcon.hide();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     * @private
+     */
+    _render: function () {
+        this._super.apply(this, arguments);
+
+        if (this.mode !== 'edit') {
+            this.$el.html(this._formatText(this.$el.text()));
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Used by MailEmojisMixin, check its document for more info.
+     *
+     * @private
+     */
+    _getTargetTextInput() {
```

Comment:
I would name it _getTargetTextElement as element could be input text or textarea